data/org.freedesktop.portal.InputCapture.xml

```diff
@@ -56,44 +56,55 @@
       This documentation describes version 1 of this interface.
    -->
   <interface name="org.freedesktop.portal.InputCapture">
-    <!--
-        CreateSession:
-        @parent_window: Identifier for the application window, see :doc:`window-identifiers`
+     <!--
+        CreateDraftSession:
         @options: Vardict with optional further information
-        @handle: Object path for the :ref:`org.freedesktop.portal.Request` object representing this call
+        @session_handle: Object path for the created :ref:`org.freedesktop.portal.Session` object
 
-        Create a capture input session. A successfully created session can at
+        Creates a draft capture input session. A successfully created session can at
         any time be closed using :ref:`org.freedesktop.portal.Session.Close`, or may
         at any time be closed by the portal implementation, which will be
         signalled via :ref:`org.freedesktop.portal.Session::Closed`.
 
         Supported keys in the @options vardict include:
 
-        * ``handle_token`` (``s``)
-
-          A string that will be used as the last element of the @handle. Must be a valid
-          object path element. See the :ref:`org.freedesktop.portal.Request` documentation for
-          more information about the @handle.
-
         * ``session_handle_token`` (``s``)
 
           A string that will be used as the last element of the session handle. Must be a valid
           object path element. See the :ref:`org.freedesktop.portal.Session` documentation for
           more information about the session handle.
 
+        This method was added in version 2 of this interface.
+    -->
+    <method name="CreateDraftSession">
+      <annotation name="org.qtproject.QtDBus.QtTypeName.In1" value="QVariantMap"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="session_handle" direction="out"/>
+    </method>
+    <!--
+        InitSession:
+        @parent_window: Identifier for the application window, see :doc:`window-identifiers`
+        @options: Vardict with optional further information
+        @session_handle: Object path for the :ref:`org.freedesktop.portal.Request` object representing this call
```

Comment:
```suggestion
        @handle: Object path for the :ref:`org.freedesktop.portal.Request` object representing this call
```
This one needs to go through a request.

The TLDR regarding requests: dbus times out faster than the user clicks so the workaround for the portals is that calls immediately return the object path for a temporary object (a `Request`) and that request then triggers a signal when the user is done. That request signal is the `Response` and contains the data we actually one (the `the following results get returned...` bit). The doc has a bit more details but the short summary is: anything that requires user interaction needs to go through a Request.

This doesn't apply to the impls since we can control the timeouts there so they don't need the request dance.

Reply:
Updated